util/stringutil/string_util.go

```diff
@@ -252,17 +252,17 @@ func Copy(src string) string {
 	return string(hack.Slice(src))
 }
 
-// stringerFunc defines string func implement fmt.Stringer.
-type stringerFunc func() string
+// StringerFunc defines string func implement fmt.Stringer.
```

Comment:
I think it's better to define a cachable struct `struct{cacheStr string; fn func() string}`, the performance will be improved if the `StringFunc` will be called many times.

Reply:
This is generally not used multiple times. Maybe we can improve it later.